spec/models/game_spec.rb

```diff
@@ -0,0 +1,137 @@
+describe Game do
+  subject { described_class.new(frame_class: frame_class, score_board: score_board) }
+
+  let(:frame) { instance_double(Frame, :frame, add_roll: nil, over?: false) }
+  let(:frame_class) { class_double(Frame, :frame_class, new: frame) }
+  let(:score_board) { class_double(ScoreBoard, :score_board, total_score: nil) }
+
+  describe '#score' do
+    it 'calls on score_board' do
+      expect(score_board).to receive(:total_score).once
+      subject.score
+    end
+  end
+
+  describe '#current_frame' do
+    it 'holds an instance of frame' do
+      expect(subject.current_frame).to be frame
+    end
+
+    it 'becomes a new frame after frame is over' do
+      allow(frame).to receive(:over?).and_return(true)
+      expect(frame_class).to receive(:new).twice
+      subject.add_roll(3)
+    end
```

Comment:
I feel like this is still testing state/implementation. The reason is that how many times you call new on state is really down to your implementation. You could choose to do it after a roll has happened, or only before you add a roll. I don't think the tests should care.